pdns/dnsdist.hh

```diff
@@ -262,7 +262,75 @@ struct DNSDistStats
   };
 };
 
+// Keeps additional information about metrics
+struct MetricDefinition {
+  MetricDefinition(std::string description, std::string prometheusType) {
+    this->description = description;
+    this->prometheusType = prometheusType;
+  }
+ 
+  MetricDefinition() = default;
+
+  // Metric description
+  std::string description;
+  // Metric type for Prometheus
+  std::string prometheusType;
+};
+
+struct MetricDefinitionStorage {
+  // Return metric definition by name
+  bool getMetricDetails(std::string metricName, MetricDefinition& metric) {
```

Comment:
`metricName` should really be a `const` reference here. Also the body of the method seems to have some indentation issues.

Reply:
Done, good point. Copying std::string isn't a good idea.